core/embed/rust/src/ui/layout/util.rs

```diff
@@ -167,10 +168,21 @@ pub extern "C" fn upy_disable_animation(disable: Obj) -> Obj {
     unsafe { try_or_raise(block) }
 }
 
-pub fn get_user_custom_image() -> Result<Gc<[u8]>, Error> {
+/*pub fn get_user_custom_image() -> Result<Gc<[u8]>, Error> {
     let len = get_avatar_len()?;
     let mut data = Gc::<[u8]>::new_slice(len)?;
     // SAFETY: buffer is freshly allocated so nobody else has it.
     load_avatar(unsafe { Gc::<[u8]>::as_mut(&mut data) })?;
     Ok(data)
+}*/
+
+pub fn get_user_custom_image() -> Result<BinaryData<'static>, Error> {
+    let len = get_avatar_len()?;
+    let mut data = Gc::<[u8]>::new_slice(len)?;
+    // SAFETY: buffer is freshly allocated so nobody else has it.
+    load_avatar(unsafe { Gc::<[u8]>::as_mut(&mut data) })?;
+    // Convert to mp_obj_bytes
+    let bytes = Obj::try_from(data.as_ref())?;
```

Comment:
please cherry-pick https://github.com/trezor/trezor-firmware/commit/79f70fc6908a392f6a733512c0ac5521b3bc6d0a and https://github.com/trezor/trezor-firmware/commit/333595ced741d146e32ca3f59db5594b919d8e3e to fix

Reply:
cherry picked, see c1eb4efffc522e23ea5683c6365159551506beeb and 22a0358a86c84ef7a4f3f36df9d967242d251915

Finally, I have modified the `len()` function to avoid using `data()`, since it is temporary and will soon be deleted. See d40f197ac910b208624e52b0459f8b60b3fcaafd